holopy/scattering/calculations.py

```diff
@@ -23,20 +23,21 @@
 """
 
 from holopy.core.holopy_object import SerializableMetaclass
-from holopy.core import Optics, Schema, Image, VectorGrid, interpret_args
+from holopy.core.metadata import vector, Image, optical_parameters
 from holopy.core.tools import dict_without, is_none
 from holopy.scattering import Mie, Multisphere, Sphere, Spheres
 from holopy.scattering.theory import dda
 from holopy.scattering.errors import AutoTheoryFailed, MissingParameter
 
 import numpy as np
+from xarray.ufuncs import square
 
 def check_schema(schema):
```

Comment:
All uses of check_schema have been removed in this PR, although I'm not sure the functionality has been maintained. Goal is that someone can call scattering functions with optical parameters specified in either schema or explicitly as scattering function arguments, but they get a MissingParameter if not specified in either place.

In any case, check_schema should be either used in the scattering functions or deleted.